src/mina-signer/tests/rosetta.test.ts

```diff
@@ -19,10 +21,46 @@ describe('Rosetta', () => {
     "stake_delegation": null
   }`;
 
+  // NOTE: copied from the test-vectors legacy signatures file
+  // RE: the rosettaTnxMockSignature above does not correctly convert
+  const legacySignatureJson = {
+    field: '2290465734865973481454975811990842289349447524565721011257265781466170720513',
+    scalar: '174718295375042423373378066296864207343460524320417038741346483351503066865',
+  };
+
   beforeAll(async () => {
     client = new Client({ network: 'mainnet' });
   });
 
+  it('generates a valid signature from a hex string', () => {
+    let { r, s } = signatureFromHex(rosettaTnxMockSignature);
+    expect(r).not.toBeNaN();
+    expect(r).not.toBeFalsy();
+    expect(s).not.toBeNaN();
+    expect(s).not.toBeFalsy();
+
+    // TODO we could go further and use validate method on the return?
+    // or leave as is with the assumption that the above hex is known to be valid
+
+    // what would the correct method be to verify?
+  });
+
+  it('generates a valid hex string from a signature', () => {
+    // leaving this in place ATM as a comparison to what appears to be simply an incorrect assumption
+    // i.e -> the mock signature may be testnet vs main or ...
+
+    // let signature = signatureFromHex(rosettaTnxMockSignature);
+    // let hex = signatureToHex(signature);
+    // fails
+    // expect(hex).toEqual(rosettaTnxMockSignature);
+
+    let hex = signatureJsonToHex(legacySignatureJson);
+    let sig = signatureFromHex(hex);
+
+    expect(sig.r).toEqual(BigInt(legacySignatureJson.field));
+    expect(sig.s).toEqual(BigInt(legacySignatureJson.scalar));
+  });
+
```

Comment:
interestingly this passes as i had expected the in-place rosetta mock to pass. this brings up the possible case that the code is, indeed, correct and that the assumption about the rosetta mock was incorrect (possibly)

there is an alternative to lines 60 and 61 that do not require knowledge of what `Signature.fromJSON` does - that is to also `import { Signature } from ...` and convert via `Signature.fromJSON(legacySignatureJson)` then directly compare to the `sig` from line 58. i don't think it's needed as it is just doing `BigInt(...)` like the test above but - worth pointing out as it might read better 🤷

Reply:
@mitschabaude see the pushed change with test-vector and that it does pass

Reply:
Before you tested fromHex -> toHex and now you're going toHex -> fromHex. I would throw in another -> toHex plus consistency check before concluding that the test works now

Reply:
yeah, i need a hex signature already existing that is known to be valid. or confirmation that the in place one is - thus proving there's some cryptog issue at play. 

i figured i'd search for another to compare, or get more info on the existing one

Reply:
Just assume that the one in here is valid

Reply:
And test toHex on your working test case